### PR TITLE
Frontload estimate usage

### PR DESCRIPTION
**Pull request by quis**
We get a bunch of requests to go live where people have told us they're going to send email but there is no email reply-to address present.

These come from 2 scenarios:

1. when there are email templates, and no reply to address – but they  ignore the checklist
2. when there are no email templates (yet) but they provide anticipated  volumes for email

At the moment we only auto-check for a reply to address when they have email templates. And because the question about anticipated volumes follows the checklist, you'll get a checklist that passes (reply addresses not required as no templates present) - but your future intent that differs (reply address IS required because you have anticipated volumes).

So let’s bring the request for anticipated volumes into the checklist, that way we can dynamically add the requirement for a reply to address if they say they will send email but don't have templates yet (later, not as part of this PR).

We should begin storing it in the database against the service to stop people having to re-enter it each time they try to complete the go live screens.

This also means moving the ‘consent to research question’ along with the questions about volume, because
- we want people to answer both before going live
- we don’t want to clutter up the summary page by asking questions there too

image

image

### CODE REVIEW

File: app/main/forms.py [hunk 1]
```diff
@@ -173,6 +173,52 @@ def __call__(self, **kwargs):
...
 
+class ForgivingIntegerField(StringField):
+
+    POSTGRES_MAX_INT = 2147483647
+
+    def process_formdata(self, valuelist):
+
+        if valuelist:
+
+            valuelist[0] = valuelist[0].replace(',', '')
```
Comment (leohemsted) [inline app/main/forms.py]: do we also want to replace spaces?

thinking of french number system type things, where you have `5 000 000`

File: app/main/forms.py [hunk 2]
```diff
@@ -173,6 +173,52 @@ def __call__(self, **kwargs):
...
+
+        return super().process_formdata(valuelist)
+
+    def pre_validate(self, form):
+
+        if self.data:
+            error = None
+            try:
+                if int(self.data) > self.POSTGRES_MAX_INT:
+                    error = 'Must be less than {:,.0f}'.format(self.POSTGRES_MAX_INT)
```
[the remark this answers is not in the document]
Comment (quis) [reply app/main/forms.py]: Rounded down to 2,000,000,000 in 4dd1a358

File: app/main/forms.py [hunk 3]
```diff
@@ -173,6 +173,52 @@ def __call__(self, **kwargs):
...
+
+            valuelist[0] = valuelist[0].replace(',', '')
+
+            try:
+                valuelist[0] = int(valuelist[0])
+            except ValueError:
+                pass
+
+            if valuelist[0] == '':
+                valuelist[0] = 0
```
[the remark this answers is not in the document]
Comment (quis) [reply app/main/forms.py]: Made it so it doesn’t re-format or zero-populate if there are errors in 61ac7fa0

File: app/templates/views/service-settings/estimate-usage.html [hunk 4]
```diff
@@ -1,35 +1,36 @@
...
-      <h1 class="heading-large">Request to go live</h1>
-
+  <div class="grid-row">
+    <div class="column-whole">
+      {% if not form.at_least_one_volume_filled %}
+        {% call banner_wrapper(type='dangerous') %}
+          <h1 class='banner-title'>
+            no things supplied
+          </h1>
+          <p>Tell us some things</p>
```
[the remark this answers is not in the document]
Comment (quis) [reply app/templates/views/service-settings/estimate-usage.html]: Up to date content added in cdf437c8

File: app/main/forms.py (hunk 1 above)
Comment (quis) [reply app/main/forms.py]: Done in a24c853d

**Comment by quis** [comment]
With latest content:

image

image

**Comment by quis** [comment]
Making the clickable areas full-width, and the task list indicators line up:

image

**Pull request merged**